src/environmentd/src/lib.rs

```diff
@@ -219,6 +222,9 @@ pub struct Config {
     // === Testing options. ===
     /// A now generation function for mocking time.
     pub now: NowFn,
+    /// When waiting for leader promotion, the server will send the socket addr of
+    /// the internal http server
+    pub waiting_on_leader_promotion: Option<Arc<oneshot::Sender<SocketAddr>>>,
```

Comment:
You generally shouldn't wrap a oneshot channel in an `Arc`. oneshot's are used so a single entity can send a single message, if you have an `Arc` that implies multiple things need access to the channel, which breaks the paradigm of a oneshot. If this is a threading/`Send` issue, we should be able to move ownership of the `oneshot::Sender` into whatever needs it.

If truly multiple things needs the ability to send data, we should use an mpsc of size 1.

Reply:
I agree it's awkward. I made it an Arc b/c Config (and the Config in environmentd/tests/util.rs) is Clone and that is used in the tests. It's not only used by a single entity at a time

Reply:
I played around with a Rc which almost works but the sqllogictest needs Send on Config b/c it spawns a thread that starts a server. In order to satisfy Clone i would need to return some empty thing (like an Option-like thing that returned None on clone) which seems to against the whole spirit of Clone in the first place. Another idea is to write another function that's similar to Clone but empties this variable... but i can't think of a good name and it's a lot of boilerplate. Ideas appreciated

Reply:
Blargh yeah this is a tough one! I think the best thing to do here is to use an `mpsc::Sender`, which is clone-able and send-able. The only other approach I can think of is to make `Config` not clone-able, I'm not sure how feasible that is though

Reply:
Cool, mpsc::Sender worked nicely

Reply:
Nice!